src/flash.sol

```diff
@@ -54,16 +62,21 @@ contract DssFlash is ReentrancyGuard {
     }
 
     // --- Mint ---
-    function mint(address _receiver, uint256 _amount, bytes calldata _params) external {
+    function mint(
+        address _receiver,      // address of conformant IFlashMintReceiver
+        uint256 _amount,        // amount to flash mint [rad]
+        bytes calldata _data    // calldata
+    ) external lock {
         require(_amount > 0, "DssFlash/amount-zero");
         require(_amount <= line, "DssFlash/ceiling-exceeded");
+        require(_data.length > 0, "DssFlash/empty-calldata");
 
         IFlashMintReceiver receiver = IFlashMintReceiver(_receiver);
 
         vat.suck(address(this), address(_receiver), _amount);
         uint256 fee = rmul(_amount, toll);
 
-        receiver.execute(_amount, fee, _params);
+        receiver.execute(_amount, fee, _data);
 
         require(vat.dai(address(this)) >= add(_amount, fee), "DssFlash/insufficient-payback");
```

Comment:
Would recommend changing the `>=` here to `==`. There is no further accounting for anything in excess of `amount + fee`, so any excess balance would remain in this contract.

The `IFlashMintReciever` should return just amount+fee, and any excess should be dealt with in the `execute`

Reply:
I was actually thinking of changing the amount sent to the vow to be whatever excess is sent beyond the fee. I doubt this will be used, but it leaves the door open to tipping, and all else being equal I don't see why not.

Reply:
Oh, yeah, I must not have written this change.  I too recommend`==` here @hexonaut.

Reply:
I've since modified the code to send any excess beyond the fee to vow. Is there any reason to cap it at the fee amount vs just sending whatever is sent? I guess the subtraction takes slightly more gas, but other then that it should be about the same.

Reply:
It's your choice, but I'd cap at the fee. I can see the developer of a receiver forgetting to account for any excess, and losing their gains to the vow. The require is kind of a sanity check that the `IFlashMintReceiver` is coded to specifications and will undo the trade to prevent them from losing more that they expected. I wouldn't be concerned about the protocol accepting tips.

Reply:
Ok you've convinced me. I'll make the change.